examples/ASViewController/Sample/ViewController.m

```diff
@@ -0,0 +1,100 @@
+/* This file provided by Facebook is for non-commercial testing and evaluation
+ * purposes only.  Facebook reserves all rights not expressly granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#import "ViewController.h"
+#import "ASTableNode.h"
+
+#import "DetailViewController.h"
+
+@interface ViewController () <ASTableDataSource, ASTableDelegate>
+@property (nonatomic, copy) NSArray *imageCategories;
+@property (nonatomic, strong, readonly) ASTableNode *tableNode;
+@end
+
+@implementation ViewController
+
+
+#pragma mark - Lifecycle
+
+- (instancetype)init
+{
+    self = [super initWithNode:[ASTableNode new]];
+    if (self == nil) { return self; }
+    
+    _imageCategories = @[@"abstract", @"animals", @"business", @"cats", @"city", @"food", @"nightlife", @"fashion", @"people", @"nature", @"sports", @"technics", @"transport"];
+    
+    return self;
+}
+
+- (void)dealloc
+{
+    self.tableNode.delegate = nil;
+    self.tableNode.dataSource = nil;
+}
+
+
+#pragma mark - UIViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.title = @"Image Categories";
+    
+    self.tableNode.delegate = self;
```

Comment:
Just do this in -init.  It won't trigger view creation.  Because of the covariant, you can do self.node.delegate without casting to tablenode.